src/SpineAnimationRenderer.ts

```diff
@@ -333,51 +370,23 @@ export class SpineAnimationRenderer extends Renderer {
     this._dirtyUpdateFlag &= ~type;
   }
 
-  /**
-   * @internal
-   */
-  _onWorldVolumeChanged(): void {
-    this._dirtyUpdateFlag |= RendererUpdateFlags.WorldVolume;
-  }
-
-  private _initialize() {
-    this._applyDefaultConfig();
-    this._dirtyUpdateFlag |= SpineAnimationUpdateFlags.InitialVolume;
-    this._state.addListener({
-      start: () => {
-        this._onAnimationStart();
-      },
-      complete: (entry: TrackEntry) => {
-        this._onAnimationComplete(entry);
-      },
-    });
-    this.update(0);
-  }
-
-  private _onAnimationStart(): void {
-    this._dirtyUpdateFlag |= SpineAnimationUpdateFlags.AnimationVolume;
-  }
-
-  private _onAnimationComplete(entry: TrackEntry): void {
-    if (!entry.loop) {
-      this._setDirtyFlagFalse(SpineAnimationUpdateFlags.AnimationVolume)
-    }
-  }
-
   private _clearMaterialCache(): void {
-    this._materials.forEach((item) => {
-      const texture = item.shaderData.getTexture('material_SpineTexture');
-      const blendMode = getBlendMode(item);
+    const materialCache = SpineAnimationRenderer._materialCache;
+    const { _materials: materials } = this;
+    for (let i = 0, len = materials.length; i < len; i += 1) {
+      const material = materials[i];
+      const texture = material.shaderData.getTexture("material_SpineTexture");
+      const blendMode = getBlendMode(material);
       const key = `${texture.instanceId}_${blendMode}`;
-      SpineAnimationRenderer._materialCache.delete(key);
-    });
+      materialCache.delete(key);
+    }
```

Comment:
Very unreasonable material design！

Reply:
The modification of materials involves the entire design of custom materials for Spine, which requires significant changes. This PR will not address this point for now. Really thank you for your understanding.